### ISSUE TITLE
`prefer-noop` shouldn't check for empty asynchronous functions

### ISSUE THREAD

**Issue by eTorAken** [wix/eslint-plugin-lodash#132]
I think there is a bug with the `prefer-noop` rule. It warns for synchronous empty functions but also for asynchronous empty functions. As `_.noop` represents a synchronous function returning `undefined`, I think this rule should not check for generator nor async functions (sad there is no `_.asyncNoop` method in Lodash).

I'm running `v2.2.5` version with `lodash/prefer-noop` set to `warn`. It throws a warning in this code but I think should not:
```jsx
import createRouter from 'koa-router';

const router = createRouter().get('/ping', function* $otherRoute() {
  // No need to yield next when testing `/ping` route.
});
```

**Comment by ganimomer** [comment]
Fixed in `v2.3.4`.
